cmd/ci_status.go

```diff
@@ -21,70 +40,177 @@ var ciStatusCmd = &cobra.Command{
+        var (
+                printer *color.Color
+        )
+        pipelineId := jobs[0].Pipeline.ID
+        pipeline, _, err := lab.Client().Pipelines.GetPipeline(pid, pipelineId)
+        if err != nil {
+                log.Fatal(errors.Wrap(err, "failed to get pipeline information"))
+        }
+        for {
+                if !summaryOnly {
+                  for _, job := range jobs {
+                            if noSkipped && job.Status == "skipped" {
+                                    continue
+                            } else if onlyFailures && job.Status != "failed" {
+                                    continue
+                            } else if noCreated && job.Status == "created" {
+                                    continue
+                            } else {
+                                    printer = statusColor(job.Status)
+                                    printer.Fprintf(w, jobFormat, job.Stage, job.Name, job.Status, job.ID)
+                            }
+                  }
+                }
+                if !wait {
+                        break
+                }
+                pl, _, err := lab.Client().Pipelines.GetPipeline(pid, pipelineId)
+                if err != nil {
+                        log.Fatal(errors.Wrap(err, "failed to get pipeline information"))
+                }
+                pipeline = pl
+                if pipeline.Status != "pending" && pipeline.Status != "running" {
+                        break
+                }
+                fmt.Fprintln(w)
+        }
+
+        fmt.Fprintf(w, pipelineStatus(pipeline, jobs))
+        if wait && pipeline.Status != "success" {
+                os.Exit(1)
+        }
+        w.Flush()
+}
+
+func pipelineStatus(pipeline *gitlab.Pipeline, jobs []*gitlab.Job) string {
+        return fmt.Sprintf("\nPipeline Status:\t%s\n%s\n\n%s\n",
+                statusColor(pipeline.Status).Sprintf(pipeline.Status), timeMessage(pipeline), jobSummary(jobs))
+}
+
+func jobSummary(jobs []*gitlab.Job) string {
+        numPassed := 0
+        totalJobs := 0
+        numQueued := 0
+        numFailed := 0
+
+        for _, job := range jobs {
+                totalJobs++
+                if (job.Status == "success") {
+                  numPassed++
+                }
+                if (job.Status == "created") {
+                  numQueued++
+                }
+                if (job.Status == "failed") {
+                  numFailed++
+                }
+        }
+
+        return fmt.Sprintf("total\tpassed\tfailed\tqueued\n%d\t%d\t%d\t%d",
+                           totalJobs, numPassed, numFailed, numQueued)
+}
+
+func timeMessage(pipeline *gitlab.Pipeline) string {
+        if pipeline.Status == "pending" {
+          return fmt.Sprintf("created at %v", pipeline.CreatedAt)
+        } else if pipeline.Status == "running" {
+          return fmt.Sprintf("started at %v", pipeline.StartedAt)
+        } else {
```

Comment:
Suggest removed the else's here and above since you are already returning based on the if cases